plugins/modules/system/disk_facts.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+#
+# Copyright 2021 VMware, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: disk_facts
+version_added: "2.5.0"
+short_description: Get disk facts in the linux OS
+description:
+    - This module is to get disk facts like free space, used percentage, mount location etc in linux Ansible controller/controlled machines.
+requirements:
+  - psutil(python module)
+options:
+    name:
+        description:
+            - This is the disk name to query
+        required: false
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter the output
+        required: false
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan(@saranyasridharan), Swetha M, Nagendra Kini, Anup Kumar, Mehmoodkha Pathan
+'''
+
+EXAMPLES = '''
+- name: Get disk facts
+  community.general.disk_facts:
+
+- name: Get Disk facts for Particular disk info
+  community.general.disk_facts:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_facts:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_facts:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/disk1s1': {
```

Comment:
I don't think this is what a facts module should return. This defines a global fact named `/dev/disk1s1`.

I'm currently thinking it would be better to have this as an `_info` module. Otherwise you have to create proper fact variables out of the disk names, like convert `capacity_percentage` in `/dev/disk1s1` to `ansible_capacity_percentage_dev_disk1s1` or something like that.

Reply:
@felixfontein Does this require renaming module and files also ?

Reply:
Yes. Also you must not return `ansible_facts` in that case.